src/modules/AddOnProduct.rb

```diff
@@ -1463,6 +1463,15 @@ def AddPreselectedAddOnProducts(filelist)
           priority = one_product.fetch("priority", -1).to_i
           prodname = one_product.fetch("name", "")
           check_name = one_product.fetch("check_name", true)
+          confirm_license = if Mode.auto
```

Comment:
You can simplify it to an one liner :wink:

```ruby
# the default value in AutoYaST is false, otherwise it is true
confirm_license = one_product.fetch("confirm_license", !Mode.auto)
```